src/CodeGen_D3D12Compute_Dev.cpp

```diff
@@ -371,7 +426,38 @@ void CodeGen_D3D12Compute_Dev::CodeGen_D3D12Compute_C::visit(const Load *op) {
                 rhs << cut.str();
             }
         }
-#endif
+        return rhs;
+    }
+};
+
+void CodeGen_D3D12Compute_Dev::CodeGen_D3D12Compute_C::visit(const Load *op) {
+    user_assert(is_one(op->predicate)) << "Predicated load is not supported inside D3D12Compute kernel.\n";
+
+    // elements in a threadgroup shared buffer are always 32bits:
+    // must reinterpret (and maybe unpack) bits...
+    if (groupshared_allocations.contains(op->name)) {
+        ostringstream rhs;
+        internal_assert(allocations.contains(op->name));
+        // no ramps when accessing shared memory...
+        Expr ramp_base = is_ramp_one(op->index);
+        internal_assert(!ramp_base.defined());
```

Comment:
I don't think you need the ramp asserts at all, if you're also asserting op->type.lanes() == 1

Reply:
Yeah, I think you're right.
At the same time, having the additional `!ramp_base.defined()` assertion adds a bit more clarity to the intent of these asserts than simply asserting on `op->type.lanes() == 1`